libs/portal-integration-angular/src/lib/services/portal-dialog.service.ts

```diff
@@ -1,8 +1,424 @@
+ *   }
+ *
+ *   ocxDialogButtonClicked(state: DialogState<string>) {
+ *     // here you can do any operations you desire
+ *     // such as form validation
+ *     // api calls and so on
+ *     if (state.button === 'primary') {
+ *       // send form data to server
+ *       this.apiService.postInput(state.result, ...).pipe(
+ *         // map response to boolean meaning if call was successfull
+ *       )
+ *     } else {
+ *       // clear input
+ *       return false // don't want to close the dialog, only to clear it
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * ## PortalDialogService call
+ * ```
+ * portalDialogService.openDialog(title, { type: MyInputComponent }, "Send", "Clear").subscribe((result: DialogState<string>) => {
+ * // behavior to be fired when dialog closes
+ * })
+ * ```
+ */
+export interface DialogButtonClicked<T = unknown> {
+  ocxDialogButtonClicked(state: DialogState<T>): Observable<boolean> | Promise<boolean> | boolean | undefined
+}
 
 /**
- * This is a bit of a hack. For some reason, providing MessageService does not work correctly across federated modules
+ * Object containing component type to be displayed and inputs to populate the component.
+ *
+ * @example
+ *
+ * ```
+ * ⁣@Component({template: `<h1>{{content}}</h1>`})
+ * export class MyComponent {
+ *   ⁣@Input() content: string = ''
+ * }
+ * const myComponent = {
+ *   type: MyComponent,
+ *   inputs: {
+ *     content: 'My header content',
+ *   },
+ * }
+ * ```
  */
-@Injectable({ providedIn: 'root' })
-export class PortalDialogService extends DialogService {}
+// eslint-disable-next-line @typescript-eslint/no-unnecessary-type-constraint
+type Component<T extends unknown> = unknown extends T
+  ? {
+      type: Type<any>
+      inputs?: Record<string, unknown>
+    }
+  : {
+      type: Type<DialogResult<T>>
+      inputs?: Record<string, unknown>
+    }
+
+/**
+ * Object containing information about clicked button ('primary' or 'secondary') and displayed component state captured on button click (only if component implements {@link DialogResult} interface)
+ */
+export type DialogState<T> = {
+  button: 'primary' | 'secondary'
+  result: T | undefined
+}
+
+@Injectable({ providedIn: 'any' })
+export class PortalDialogService {
+  constructor(private dialogService: DialogService, private translateService: TranslateService) {}
+
+  /**
+   * @deprecated
+   */
+  open(componentType: Type<any>, config: DynamicDialogConfig): DynamicDialogRef {
+    if (isDevMode()) {
+      console.warn('You are using a deprecated method to display a dialog. Please move to the new one')
+    }
+    return this.dialogService.open(componentType, config)
+  }
+
+  /**
+   * Opens dialog with a component or message to display and one or two buttons. This method allows you to customize the dialog using parameters and by implementic specific interfaces via component to be displayed.
+   *
+   * @param title Dialog title
+   * @param componentOrMessage Dialog content to display
```

Comment:
Either a component or a translation key of a message with optional parameters and icon to be displayed next to the message